qiskit/providers/models/pulsedefaults.py

```diff
@@ -94,44 +98,335 @@ class Command(BaseModel):
+                                 "{qubits}.".format(op=operation,
+                                                    qubits=self.qubits_with_op(operation)))
+            raise PulseError("Operation '{op}' is not defined for this "
+                             "system.".format(op=operation))
+
+    def get(self,
+            operation: str,
+            qubits: Union[int, Iterable[int]],
+            *params: List[Union[int, float, complex]],
+            **kwparams: Dict[str, Union[int, float, complex]]) -> Schedule:
         """
-        return CmdDef.from_defaults(self.cmd_def, self.pulse_library)
+        Return the defined Schedule for the given operation on the given qubits.
+
+        Args:
+            operation: Name of the operation.
+            qubits: The qubits for the operation.
+            *params: Command parameters for generating the output schedule.
+            **kwparams: Keyworded command parameters for generating the schedule.
+
+        Returns:
+            The Schedule defined for the input.
+
+        Raises:
+            PulseError: If the operation is not defined on the qubits.
+        """
+        self.assert_has(operation, qubits)
+        schedule = self._ops_def[operation].get(_to_tuple(qubits))
+        if isinstance(schedule, ParameterizedSchedule):
+            schedule = schedule.bind_parameters(*params, **kwparams)
+        return schedule
+
+    def get_parameters(self, operation: str, qubits: Union[int, Iterable[int]]) -> Tuple[str]:
+        """
+        Return the list of parameters taken by the given operation on the given qubits.
+
+        Args:
+            operation: Name of the operation.
+            qubits: The qubits for the operation.
+
+        Returns:
+            The names of the parameters required by the operation.
+
+        Raises:
+            PulseError: If the operation is not defined on the qubits.
+        """
+        self.assert_has(operation, qubits)
+        return self._ops_def[operation][_to_tuple(qubits)].parameters
+
+    def add(self,
+            operation: str,
+            qubits: Union[int, Iterable[int]],
+            schedule: [Schedule, ParameterizedSchedule]) -> None:
+        """
+        Add a new known operation.
+
+        Args:
+            operation: The name of the operation to add.
+            qubits: The qubits which the operation applies to.
+            schedule: The Schedule that implements the given operation.
+
+        Returns:
+            None
+
+        Raises:
+            PulseError: If the qubits are provided as an empty iterable.
+        """
+        qubits = _to_tuple(qubits)
+        if qubits == ():
+            raise PulseError("Cannot add definition {} with no target qubits.".format(operation))
+        if not isinstance(schedule, (Schedule, ParameterizedSchedule)):
+            raise PulseError("Attemping to add an invalid schedule type.")
+        if self.has(operation, qubits):
```

Comment:
This warning could get annoying very quick, I'm not sure its necessary.

Reply:
Hmm, yeah, I guess it depends how likely it is that a user would want to overwrite a gate vs accidentally adding another gate without giving it a new name. If you want it taken out, I'm fine with that

Reply:
Yah, let's remove for now.